src/include/migraphx/compile_options.hpp

```diff
@@ -40,6 +41,8 @@ struct compile_options
 
     bool fast_math       = true;
     bool exhaustive_tune = false;
+    bool weight_streaming = false;
```

Comment:
I dont think we should have a flag to enable weight streaming, if you run out of memory then it is the only way to run the model, so it should be enabled by default.

Reply:
How would we determine if the model runs out of memory? Unless the model is first compiled and run with no weight streaming to test if there is sufficient memory, but that would also require re-compiling anyways after a failure.

Reply:
We can take the estimated scratch+literals+parameters to determine how much the memory the model will use. The estimated scratch might be an overestimate, but this is a boderline case. Its better to nder use the memory the over use it and fail. Furthermore, if the user really wants to use it on the GPU they can set the max memory larger.